### ISSUE TITLE
Unable to use compose decorator in development mode

### ISSUE THREAD

**Issue by anirban09** [fabiospampinato/overstated#8]
I created a new React app using:
`npx create-react-app my-app --typescript`

And added overstated with:
`yarn add overstated@2.0.0-beta.4`

But I was unable to use the `compose` decorator when I ran the development server with `yarn start`. It gives this error:
`TypeError: Class constructor App cannot be invoked without 'new'`

From [this Stack Overflow thread](https://stackoverflow.com/questions/51860043/javascript-es6-typeerror-class-constructor-client-cannot-be-invoked-without-ne/51860850) and [this issue (comment)](https://github.com/facebook/create-react-app/issues/1751#issuecomment-284763003), it seems that it's because there's a mismatch between native ES6 classes and classes transpiled to ES5.

But it builds fine, with `yarn build`, because then everything is transpiled to ES5, I suppose. And it works fine when the built static files are served.

Here's a repo to reproduce:
https://github.com/anirban09/overstated-transpile-problem.git
Clone this, and run `yarn install` and `yarn start` (or `npm install` and `npm start`) to reproduce the issue.

The question is, how to use the `compose` decorator in a new React app in development mode?

**Comment by fabiospampinato** [comment]
@anirban09 I'll take a look at the provided example 👍 If the issue is with calling `compose` as a decorator you could just call it as a regular function : `compose ({ foo: Foo })( MyOtherStore )`.

**Comment by anirban09** [comment]
No, unfortunately that's not the issue, it doesn't work when using `compose` as a regular function either.

**Comment by fabiospampinato** [comment]
I'll take a closer look at this later. Perhaps in the meantime you could just pass to compose pre-instantiated stores:

```js
compose ({
  foo: new Foo ()
});
```

**Comment by anirban09** [comment]
Hey, no, the issue is in instantiating the parent store, not any of the child stores.
When running the react-scripts development server, the `compose` function fails because there, the ES5 "class" in the `compose` function is trying to use my app's parent store, which is provided as a native ES6 class.

Not sure if you could call this a bug, exactly. Guess the target environments of overstated and the react-scripts dev server are different.
If I change the `target` compiler option in overstated's tsconfig.json to `es6` and compile it, and place it in my app's node_modules, it works.

But anyone who creates a React app using create-react-app and tries to use overstated will face this problem, so there should probably be some way to mitigate it.

**Comment by fabiospampinato** [comment]
Ok so the problem comes from, as you correctly mentioned, extending an es6 class from an es5 "class", the following piece of code reproduces the problem:

```ts
// TS's extend helper
var __extends = (this && this.__extends) || (function () {
  var extendStatics = function (d, b) {
      extendStatics = Object.setPrototypeOf ||
          ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
          function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
      return extendStatics(d, b);
  };
  return function (d, b) {
      extendStatics(d, b);
      function __() { this.constructor = d; }
      d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
  };
})();
// ES6 class
class foo {}
// ES5 class
var bar = /** @class */ (function (_super) {
  __extends(bar, _super);
  function bar() {
      return _super !== null && _super.apply(this, arguments) || this;
  }
  return bar;
}(foo));
// Instantiation
console.log ( new bar () );
```

I'll check if there's any clean solution to this, but I doubt it.

I think most probably I'll have to publish the library targeting es6, in a way that if you want to the bundler could then itself transpile it down to es5, that should work I think.

**Comment by fabiospampinato** [comment]
This is fixed in v2.0.0-beta.5

In order to fix this I had to resort to `eval` unfortunately.

Thanks for this issue report!
